repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeRoutingServlet.scala

```diff
@@ -0,0 +1,362 @@
+  private val domains = new AtomicReference[Option[Clusters]](None) // TODO: Remove this when routing to a cluster is removed.
+  private var initialized = false
+
+  override def init(): Unit = {
+    logger.info("{}:{} -- Reticulating Splines - Initializing Repose Routing Servlet", clusterId, nodeId)
+    configurationService.subscribeTo(
+      SystemModelConfigurationFilename,
+      getClass.getResource("/META-INF/schema/system-model/system-model.xsd"),
+      this,
+      classOf[SystemModel]
+    )
+  }
+
+  override def destroy(): Unit = {
+    configurationService.unsubscribeFrom(SystemModelConfigurationFilename, this)
+    logger.info("{}:{} -- Obfuscated Quigley Matrix - Destroying Repose Routing Servlet", clusterId, nodeId)
+  }
+
+  override def configurationUpdated(configurationObject: SystemModel): Unit = {
+    //Set the current system model, and just update the nodes.
+    val interrogator = new SystemModelInterrogator(clusterId, nodeId)
+    localCluster.set(Option(interrogator.getLocalCluster(configurationObject).orElse(null)))
+    localNode.set(Option(interrogator.getLocalNode(configurationObject).orElse(null)))
+    defaultDestination.set(Option(interrogator.getDefaultDestination(configurationObject).orElse(null)))
+    destinations.set(localCluster.get().map(_.getDestinations) match {
+      case Some(dests) => getDestinations(dests.getEndpoint.asScala.toList) ++ getDestinations(dests.getTarget.asScala.toList)
+      case None => Map.empty[String, Destination]
+    })
+    domains.set(Option(new Clusters(configurationObject)))
+
+    if (logger.underlying.isDebugEnabled) {
+      localCluster.get().foreach { cluster =>
+        val clusterId: String = cluster.getId
+        //Build a list of the nodes in this cluster, just so we know what we're doing
+        val clusterNodes = cluster.getNodes.getNode.asScala.map(node => s"${node.getId}-${node.getHostname}")
+        logger.debug("{}:{} - Nodes for this router: {}", clusterId, nodeId, clusterNodes)
+        val destStrings = destinations.get()
+          .filter { case (_, dest) => dest.isInstanceOf[DestinationEndpoint] }
+          .map { case (_, endpoint) => s"${endpoint.getId}-${endpoint.asInstanceOf[DestinationEndpoint].getHostname}" }
+        logger.debug("{}:{} - Destinations for this router: {}", clusterId, nodeId, destStrings)
+      }
+    }
+    initialized = true
+
+    def getDestinations(destList: List[Destination]): Map[String, Destination] = {
+      destList.iterator.map(dest => dest.getId -> dest).toMap
+    }
+  }
+
+  override def isInitialized: Boolean = {
+    initialized
+  }
+
+  override def service(req: HttpServletRequest, resp: HttpServletResponse): Unit = {
+    if (!initialized) {
+      logger.error(NotInitializedMessage)
+      trySendError()
+    } else {
+      logger.trace("{} processing request...", this.getClass.getSimpleName)
+      Try(doRoute(new HttpServletRequestWrapper(req), resp)) recover {
```

Comment:
I haven't really made up my mind about when to use `Try` vs `try/catch`. In this case, I lean toward using `try/catch` since you aren't doing anything with the return value of `doRoute` (or lack thereof, really). Under the hood, that's basically what `Try` is doing anyway, so this is just a wrapper to use a different syntax. Does that alone improve the code in some way? I'm not sure. It adds another object to the heap, but that's minimal.

🤷‍♂️

Reply:
All of the uses of `try`/`catch` were flagged by Codacy as use `util.Try`. This seems to be the Scala way, so I did it.